### ISSUE TITLE
Ant Design Table a11y Issues

### ISSUE THREAD

**Issue by ericbeijer** [ant-design/ant-design#53819]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/ant-design-table-a11y-issues-r55tsg)

### Steps to reproduce

When setting `scroll.y` to a number (see upper table vs. lower table, make sure the tables have a decent amount of horizontal scroll bar) ...

- ... horizontal scrolling does not work when the mouse is over the header
- ... tabbing through the header does not scroll the table body (a11y issue)
  - also using the enter key to set the sorting scrolls the header to the position of the body
- ... the arrow keys cannot be used to scroll the table when focus is on a header cell (a11y issue)

### What is expected?

When `scroll.y` is set to a number, the scroll position of the header and the body should be in sync. Also when the focus is on a header cell, the arrow keys should allow the user to scroll.

### What is actually happening?

When `scroll.y` is set to a number, tabbing through the header cells does not scroll the table body along. Also the arrow keys do not scroll when focus is on a header cell.

| Environment | Info |
| --- | --- |
| antd | 5.25.1 |
| React | 18.3.1 |
| System | Ubuntu |
| Browser | Chrome |

---
These issues seem vastly related to https://github.com/ant-design/ant-design/issues/11194.

**Comment by prakashk-mallow** [comment]
I'm facing the following WCAG issues from the Table component

1. XPATH -> //html/body/div[2]/main/div/div[2]/main/div/div/div/div/div/div/div/div[2]/table/tbody/tr
Focusable content should be disabled or be removed from the DOM

<img width="1169" height="377" alt="Image" src="https://github.com/user-attachments/assets/96f75d9c-53ed-400e-8a0d-aef47ba9311d" />
<img width="1169" height="377" alt="Image" src="https://github.com/user-attachments/assets/93ab68c3-449e-4148-a8f8-e57f6a48f974" />

2. XPATH -> //html/body/div[3]/main/div/div[2]/main/div/div/div/div/div/div/div/div/table
Ensure that <th> elements and elements with role=columnheader/rowheader have data cells they describe. Table data cells are missing or empty inside the 'table-header' on the Table component

<img width="1169" height="405" alt="Image" src="https://github.com/user-attachments/assets/48f110e2-d088-41c1-9227-b542d4c2cfe3" />
<img width="1169" height="405" alt="Image" src="https://github.com/user-attachments/assets/1ae908db-f87b-4256-8c9d-bc4d9ae9e863" />

<img width="1132" height="311" alt="Image" src="https://github.com/user-attachments/assets/67c1986d-1e7e-40aa-ac0b-1feb9836d64b" />

How can we fix the above issues?
